File: comicop-frontend/src/services/addressService.ts
import api from './api'

export interface AddressDto {
  addressId: number
  fullName: string
  phone: string
  street: string
  provinceCode: string
  provinceName: string
  wardCode: string
  wardName: string
  fullAddress: string
  isDefault: boolean
}

export interface CreateAddressRequest {
  fullName: string
  phone: string
  street: string
  provinceCode: string
  provinceName: string
  wardCode: string
  wardName: string
  isDefault?: boolean
}

export const addressService = {
  getAddresses: () => api.get<AddressDto[]>('/addresses'),
  createAddress: (data: CreateAddressRequest) =>
    api.post<AddressDto>('/addresses', data),
  setDefault: (id: number) =>
    api.put<AddressDto>(`/addresses/${id}/default`),
  deleteAddress: (id: number) =>
    api.delete(`/addresses/${id}`),
}

File: comicop-backend/src/main/java/com/example/comicop/service/AddressService.java
package com.example.comicop.service;

import com.example.comicop.dto.AddressDto;
import com.example.comicop.dto.CreateAddressRequest;
import java.util.List;

public interface AddressService {
    List<AddressDto> getAddresses(Long accountId);
    AddressDto createAddress(Long accountId, CreateAddressRequest request);
    AddressDto setDefault(Long addressId, Long accountId);
    void deleteAddress(Long addressId, Long accountId);
}